### PR TITLE
Feat: Add ability to invalidate the environment

### PR DESCRIPTION
**Pull request by izeigerman**
This update allows user to invalidate the target environment by setting its expiration timestamp to now, forcing it to be garbage collected during a subsequent run of the janitor process.

CC: @z3z1ma

### CODE REVIEW

File: sqlmesh/integrations/github/cicd/controller.py [hunk 1]
```diff
@@ -407,15 +406,7 @@ def delete_pr_environment(self) -> None:
...
-        schemas = {parse_model_name(model.name)[1] for model in self._context.models.values()}
-        for schema in schemas:
-            assert schema
-            self._context.engine_adapter.drop_schema(
-                schema_name="__".join([schema, self.pr_environment_name]),
-                ignore_if_not_exists=True,
-                cascade=True,
-            )
-        return
+        self._context.invalidate_environment(self.pr_environment_name)
```
Comment (izeigerman) [inline sqlmesh/integrations/github/cicd/controller.py:409]: CC: @eakmanrq please take a look

**Review by tobymao — APPROVED** [review]

**Comment by z3z1ma** [comment]
Would `teardown` make sense here as being more intuitive to an end user? That way users can "teardown" a non-prod environment in a single command which is sort of the "button" I can imagine people looking for. 

`teardown` == `invalidate` + `janitor` execution

Thoughts?

**Comment by izeigerman** [comment]
@z3z1ma the problem with both `destroy` and `teardown` is that I think it would be misleading in this case. The environment doesn't get "destroyed" upon running the command. It's merely being marked for deletion at some point in time later. 

Additionally, we invoke the janitor process directly only when using the built-in scheduler. With Airflow the janitor runs based on a cadence configured by a user. The janitor process can be pretty expensive and long since it collects all expired environments and not just the one specified in the command, so I think it's better to keep it async.

File: sqlmesh/integrations/github/cicd/controller.py [hunk 2]
```diff
@@ -407,15 +406,7 @@ def delete_pr_environment(self) -> None:
         """
         Deletes all the schemas for a given environment by checking all the schemas used by models.
```
Comment (eakmanrq) [inline sqlmesh/integrations/github/cicd/controller.py:407]: Can you update this docstring?

**Review by eakmanrq — APPROVED** [review]

**Comment by eakmanrq** [comment]
At least form a CI/CD bot perspective, async deletion of an environment is fine IMO. It is ok if the environments sticks around for a bit until the next janitor run. I get what you are saying though @z3z1ma if there was a need for an environment to be immediately removed but I don't see that need currently.

**Pull request merged**